dataservices/dataservices.go

```diff
@@ -0,0 +1,239 @@
+			return
+		}
+
+		platformData, err := data.NewTypeBuilder(map[string]interface{}{data.UserIDField: userid, data.GroupIDField: groupID}).BuildFromDataSet(dataSet)
+		processedDataset.Dataset = platformData
+		processedDataset.Errors = err.Error()
+
+		if err.Error() != "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.WriteJson(&processedDataset)
+			return
+		}
+
+		//TODO: should this be a bulk insert?
+		for i := range platformData {
+			if err = client.dataStore.Save(platformData[i]); err != nil {
+				rest.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+		}
+
+		w.WriteJson(&processedDataset)
+		return
+	}
+	rest.Error(w, missingPermissionsError, http.StatusUnauthorized)
+	return
+}
+
+//PostBlob will save a posted blob of  data for the requested user if permissons are sufficient
+func (client *DataServiceClient) PostBlob(w rest.ResponseWriter, r *rest.Request) {
+
+	userid := r.PathParam(useridParamName)
+	log.AddTrace(userid)
+
+	if checkPermisson(r, user.Permission{}) {
+		w.WriteHeader(http.StatusNotImplemented)
+		return
+	}
+	rest.Error(w, missingPermissionsError, http.StatusUnauthorized)
+	return
+
+}
+
+//process the found data and send the appropriate response
+func process(iter store.Iterator) data.Dataset {
+
+	var chunk data.Datum
+	var all = data.Dataset{}
+
+	for iter.Next(&chunk) {
+		all = append(all, chunk)
+	}
+
+	return all
+}
+
+//GetDataset will return the requested users data set if permissons are sufficient
+func (client *DataServiceClient) GetDataset(w rest.ResponseWriter, r *rest.Request) {
+
+	log.AddTrace(r.PathParam(useridParamName))
+
+	if checkPermisson(r, user.Permission{}) {
+
+		var found struct {
+			data.Dataset `json:"Dataset"`
+			Errors       string `json:"Errors"`
+		}
+
+		userid := r.PathParam(useridParamName)
+		log.Info(useridParamName, userid)
+
+		types := strings.Split(r.URL.Query().Get("type"), ",")
+		subTypes := strings.Split(r.URL.Query().Get("subType"), ",")
+		start := r.URL.Query().Get("startDate")
+		end := r.URL.Query().Get("endDate")
+
+		log.Info("params", types, subTypes, start, end)
+
+		iter := client.dataStore.ReadAll(
+			store.Fields{data.UserIDField: userid},
+			data.InternalFields,
+		)
+		defer iter.Close()
+
+		found.Dataset = process(iter)
+
+		w.WriteJson(&found)
+		return
+	}
+	rest.Error(w, missingPermissionsError, http.StatusUnauthorized)
+	return
+}
+
+//GetData will return the requested users data point if permissons are sufficient
+func (client *DataServiceClient) GetData(w rest.ResponseWriter, r *rest.Request) {
+
+	log.AddTrace(r.PathParam(useridParamName))
+
+	if checkPermisson(r, user.Permission{}) {
+		var foundDatum struct {
+			data.Datum `json:"Datum"`
+			Errors     string `json:"Errors"`
+		}
+
+		userid := r.PathParam(useridParamName)
+		datumid := r.PathParam("datumid")
+
+		log.Info("userid and datum", userid, datumid)
+
+		foundDatum.Datum = data.Datum{}
+		foundDatum.Errors = ""
+
+		w.WriteJson(&foundDatum)
```

Comment:
Do you have an example of what the JSON output from all of the APIs looks like?  I'm particularly interested in the Errors and how that plays out.